actions/issue/add-to-project/entrypoint

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+set -euo pipefail
+shopt -s inherit_errexit
+
+# Github CLI does not currently allow interacting with the API of new-style projects.
+# Github GraphQL API guide: https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects
+
+function main() {
+  declare issue_node_id project_org project_num field_name option_name
+  declare -i project_num
+
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --token)
+        token="${2}"
+        shift 2
+        ;;
+
+      --issue-node-id)
+        issue_node_id="${2}"
+        shift 2
+        ;;
+
+      --project-org)
+        project_org="${2}"
+        shift 2
+        ;;
+
+      --project-num)
+        project_num=${2}
+        shift 2
+        ;;
+
+      --field-name)
+        field_name="${2}"
+        shift 2
+        ;;
+
+      --option-name)
+        option_name="${2}"
+        shift 2
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        echo "Unknown argument" "$@"
+        exit 1
+    esac
+  done
+
+  echo "${token}" | gh auth login --with-token
+
+  project_id=
+  field_id=
+  option_id=
+  fill_ids_from_input
+
+  project_item_id=
```

Comment:
I'd have a slight preference for `project_item_id` to be defined by the output of the `add_issue_to_proj` function (i.e. `project_item_id="$(add_issue_to_project)"` - obviously changing that function to return the new ID.

I think it's easier to keep track of variables this way, vs having them set in a more global manner and overridden inside other functions.

If you agree, would you be willing to make that change also?

Reply:
Also this will probably require you to pass that variable as an argument to the `set_proj_item_field()` function. I think that's ok. I think it makes it more obvious what functions require and what they modify.

Reply:
Done. See refactored code